From 5dfe2f0d59ba5f141dc7e9d607b68d5b1e8e57c0 Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Tue, 22 Jan 2013 15:43:56 +0800
Subject: [PATCH] at91sam9x5: mmc: save environment as a file in FAT partition.

This patch will save U-Boot environment as a file: uboot.env, in FAT partition
instead of saving it in raw sector of SD card.
Since saving environment in raw sector has risk of corrupting the SD card and
only can use very small size.
Save as a FAT file has no above limitation.

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 include/configs/at91sam9x5ek.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/include/configs/at91sam9x5ek.h b/include/configs/at91sam9x5ek.h
index b1d4baa..c10228a 100644
--- a/include/configs/at91sam9x5ek.h
+++ b/include/configs/at91sam9x5ek.h
@@ -203,11 +203,12 @@
 				"bootm 0x22000000"
 #else /* CONFIG_SYS_USE_MMC */
 /* bootstrap + u-boot + env + linux in mmc */
-#define CONFIG_ENV_IS_IN_MMC
-/* For FAT system, most cases it should be in the reserved sector */
-#define CONFIG_ENV_OFFSET	0x2000
-#define CONFIG_ENV_SIZE		0x1000
-#define CONFIG_SYS_MMC_ENV_DEV	0
+#define CONFIG_ENV_IS_IN_FAT
+#define CONFIG_FAT_WRITE
+#define FAT_ENV_INTERFACE	"mmc"
+#define FAT_ENV_FILE		"uboot.env"
+#define FAT_ENV_DEVICE_AND_PART	"0"
+#define CONFIG_ENV_SIZE		0x4000
 #endif
 
 #ifdef CONFIG_SYS_USE_MMC
-- 
1.9.1

